helix-term/src/ui/prompt.rs

```diff
@@ -253,12 +267,21 @@ impl Component for Prompt {
                 code: KeyCode::Enter,
                 ..
             } => {
-                (self.callback_fn)(cx.editor, &self.line, PromptEvent::Validate);
-                return close_fn;
+                if self.line.ends_with('/') {
```

Comment:
I was thinking we could make this more generic for all completions: if a completion is selected then `Enter` would fill it in and update completion. So after tab completing you'd press enter twice, once to confirm the completion, once to submit.

Reply:
I thought about that as well, but felt it was a bit less ergonomic for files since that's the end of completion. In Vim you can press `CTRL-d` to trigger completion, perhaps that's a better alternative?

Reply:
Ah, I didn't know about `ctrl-d`, usually I have to do some fancy stuff.

Reply:
@kevinsjoberg command completion uses the same logic (`:<tab>`), and we don't want to submit on enter there since the user might want to enter arguments after the command. That's why I was thinking Enter should simply confirm a selection without submitting

Reply:
Not sure I follow. Pressing `Tab` on `:o` would complete it to `:open`, allowing you to fill in whatever arguments you'd like. Pressing `Enter` would execute the action (just as it is now).

Reply:
Right, true

Reply:
The assumption I've made now, is that completion items ending with `/` are directories and thus should re-trigger completion. For all other use cases the behavior should be identical to what we're currently seeing. 🙂

Reply:
Yup, these are the cases I have to type some character and delete it to fix it.